src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -0,0 +1,333 @@
+        {
+            SERVICE_KERNEL_DRIVER = 0x00000001,
+            SERVICE_FILE_SYSTEM_DRIVER = 0x00000002,
+            SERVICE_ADAPTER = 0x00000004,
+            SERVICE_RECOGNIZER_DRIVER = 0x00000008,
+            SERVICE_WIN32_OWN_PROCESS = 0x00000010,
+            SERVICE_WIN32_SHARE_PROCESS = 0x00000020,
+            SERVICE_INTERACTIVE_PROCESS = 0x00000100
+        }
+
+        /// <summary>
+        /// Describes service start type.
+        /// </summary>
+        public enum ServiceStartType : uint
+        {
+            SERVICE_BOOT_START,
+            SERVICE_SYSTEM_START,
+            SERVICE_AUTO_START,
+            SERVICE_DEMAND_START,
+            SERVICE_DISABLED
+        }
+
+        /// <summary>
+        /// Describes the severity of the error, and action taken, if this service fails to start.
+        /// </summary>
+        public enum ServiceErrorControl : uint
+        {
+            SERVICE_ERROR_IGNORE,
+            SERVICE_ERROR_NORMAL,
+            SERVICE_ERROR_SEVERE,
+            SERVICE_ERROR_CRITICAL
+        }
+
+        /// <summary>
+        /// Describes the configuration information to be changed.
+        /// </summary>
+        public enum ServiceInfoLevel
+        {
+            SERVICE_CONFIG_DESCRIPTION = 1,
+            SERVICE_CONFIG_FAILURE_ACTIONS,
+            SERVICE_CONFIG_DELAYED_AUTO_START_INFO,
+            SERVICE_CONFIG_FAILURE_ACTIONS_FLAG,
+            SERVICE_CONFIG_SERVICE_SID_INFO,
+            SERVICE_CONFIG_REQUIRED_PRIVILEGES_INFO,
+            SERVICE_CONFIG_PRESHUTDOWN_INFO,
+            SERVICE_CONFIG_TRIGGER_INFO,
+            SERVICE_CONFIG_PREFERRED_NODE,
+            SERVICE_CONFIG_LAUNCH_PROTECTED = 12
+        }
+
+        /// <summary>
+        /// Changes the optional configuration parameters of a service.
+        /// </summary>
+        /// <param name="hService">
+        /// A handle to the service.
+        /// This handle is returned by the OpenService or CreateService function and
+        /// must have the <see cref="ServiceAccess.SERVICE_CHANGE_CONFIG"/> access right.
+        /// </param>
+        /// <param name="dwInfoLevel">
+        /// The configuration information to be changed.
+        /// This parameter can be one value from <see cref="ServiceStartType"/>.
+        /// </param>
+        /// <param name="lpInfo">
+        /// A pointer to the new value to be set for the configuration information. 
+        /// The format of this data depends on the value of the dwInfoLevel parameter. 
+        /// If this value is NULL, the information remains unchanged.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// If the function fails, the return value is zero
+        /// </returns>
+        [DllImport(nameof(AdvApi32), SetLastError = true)]
+        public static extern bool ChangeServiceConfig2(IntPtr hService, ServiceInfoLevel dwInfoLevel, IntPtr lpInfo);
```

Comment:
Also I think that instead of returning / Taking `IntPtr` a `SafeHandle`-derived class calling `CloseServiceHandle` should be defined and used everywhere (Except in `CloseServiceHandle` itself)